subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -543,6 +827,20 @@ int nrf_cloud_agps_process(const char *buf, size_t buf_len, const int *socket)
 	size_t parsed_len = 0;
 	uint8_t version;
 
+	err = parse_cell_location_response(buf);
```

Comment:
Just so I understand this, it looks to me like `nrf_cloud_agps_process()` handles both the older binary format for full AGPS, as well as this newer JSON format single cell response.  Wouldn't it make more sense to only call `parse_cell_location_response()` when building with `CONFIG_NRF_CLOUD_AGPS_SINGLE_CELL_ONLY` support?

Reply:
you could make a cell-based location request without enabling `CONFIG_NRF_CLOUD_AGPS_SINGLE_CELL_ONLY`.
not sure if there is a use-case for that?

Reply:
I assume you've tested this change with binary AGPS data to ensure the JSON parser doesn't misbehave.